src/ledger.ts

```diff
@@ -0,0 +1,164 @@
+// LedgerSystem class
+import BigNumber from "bignumber.js";
+import * as dataConnectors from "./data-connectors";
+import { FIFOQueue } from "./queue";
+import { sleep, groupBy } from "./utils";
+import { IPostingEntry, BaseDataConnector, IOperation, OperationStatus, OperationType} from "./base-data-connector";
+export interface IPostingEntryRequest {
+    bookId: string,
+    assetId: string,
+    value: string,
+}
+export interface IOperationRequest {
+    type: OperationType,
+    memo: string,
+    entries: IPostingEntryRequest[],
+}
+export interface ITransferRequest {
+    fromBookId: string,
+    toBookId: string,
+    assetId: string,
+    value: string,
+    memo: string,
+}
+export interface IBookRequest {
+    name: string,
+    metadata?: {},
+    restrictions?: {
+        minBalance?: string;
+    }
+}
+export interface IBookBalances {
+    [assetId: string]: string,
+}
+export class LedgerSystem {
+    private dataConnector: BaseDataConnector;
+    private postingQueue!: FIFOQueue;
+    private initPromise: Promise<void>;
+    constructor (storageConfig: {class: dataConnectors.DataConnectorType}) {
+        this.dataConnector = new dataConnectors[storageConfig.class](storageConfig);
+        this.initPromise = this.init();
+    }
+    private async init() {
+        // initialisation of the postingQueue
+        const pendingTaskQueue = await this.getPendingOperations()
+            .then((pendingOperations) => pendingOperations.map((operation) => this.getOperationTask(operation)))
+        this.postingQueue = new FIFOQueue(pendingTaskQueue);
+    }
+    private async getPendingOperations() {
+        return this.dataConnector.getAllOperationsByStatus([OperationStatus.INIT, OperationStatus.PROCESSING])
+            .then((pendingOperations) => {
+                return pendingOperations.sort((operationA: IOperation, operationB: IOperation) => operationA.id!.localeCompare(operationB.id!))
+            });
+    }
+    private getOperationTask(operation: IOperation) {
+        return async () => {
+            await this.postOperationEntries(operation.id!, operation.entries)
```

Comment:
Please don't use the word 'post'

It's confusing with 'postingQueue` and HTTP POST